### PR TITLE
Update GridPolygon.py

### PR DESCRIPTION
**Pull request by lbartoletti**
Add an overlay option

### CODE REVIEW

File: python/plugins/processing/algs/qgis/GridPolygon.py [hunk 1]
```diff
@@ -87,11 +95,15 @@ def processAlgorithm(self, progress):
...
-        originY = bbox.yMaximum()
+        originY = bbox.yMinimum()
 
         if hSpacing <= 0 or vSpacing <= 0:
             raise GeoAlgorithmExecutionException(
                 self.tr('Invalid grid spacing: %s/%s' % (hSpacing, vSpacing)))
+                
+        if hSpacing <= hOverlay or vSpacing <= vOverlay:
+            raise GeoAlgorithmExecutionException(
+                self.tr('Invalid overaly: %s/%s' % (hOverlay, vOverlay)))
```
Comment (nyalldawson) [inline python/plugins/processing/algs/qgis/GridPolygon.py]: Same typo here

**Comment by nyalldawson** [comment]
Hi @lbartoletti  - as with #3547 can you also add a unit test covering the changes introduced by this PR?

**Comment by nyalldawson** [comment]
@lbartoletti thanks for the updates. This PR has merge conflicts which prevent it being merged - can you please fix these?

**Comment by lbartoletti** [comment]
Closed, look at #3934

**Pull request closed without merging**